plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Marketing.php

```diff
@@ -4,12 +4,35 @@
 
 use Automattic\WooCommerce\Admin\Features\Features;
 use Automattic\WooCommerce\Admin\Features\OnboardingTasks\Task;
-use Automattic\WooCommerce\Internal\Admin\RemoteFreeExtensions\Init as RemoteFreeExtensions;
 
 /**
  * Marketing Task
  */
 class Marketing extends Task {
+	/**
+	 * Constructor
+	 *
+	 * @param TaskList $task_list Parent task list.
+	 */
+	public function __construct( $task_list ) {
+		parent::__construct( $task_list );
+
+		if ( $task_list->visible && !$task_list->is_hidden() && ! $this->is_complete() ) {
```

Comment:
I think we should avoid running these potentially expensive checks on every page load. Since the activated_plugin hook only fires when a plugin is actually being activated, maybe we don't need to check the task list state? Or perform these checks when the relevant plugin is actually being activated

Reply:
Good catch! Sounds good 👍 

Changed made in d99afa9aed9